### DISCUSSION TITLE
errors with "field" configuration for Multivariate MODE

### DISCUSSION THREAD

**Discussion by sarahmonet** [dtcenter/METplus#1498] [Statistical Computation] [answered]
Hi,
I'm trying to run Multivariate MODE using the attached configuration file.
[mode_config_file_v10_multivariate.docx](https://github.com/dtcenter/METplus/files/8209896/mode_config_file_v10_multivariate.docx)

However, I'm getting the error
WARNING: 
WARNING: Dictionary::lookup_dictionary() -> dictionary lookup failed for name "field"
WARNING: 
 
I don't think my "field" is formatted correctly, even though I based it off of this presentation.
![Screen Shot 2022-03-08 at 3 12 00 PM](https://user-images.githubusercontent.com/20075584/157332886-df256706-1103-49b7-8a97-21f0bf5ebd59.png)

How do I need to format "field" to run Multivariate MODE.

Thanks!
Sarah

**Comment by TaraJensen** [comment]
Sarah,

I see you are trying to run Multivariate MODE.  Your errors are not surprising because we recently determined that the MvMODE development was not merged into MET v10.0.0 as we expected.  We just recently merged the capability into the MET v10.1.0 develop branch but there were too many conflicts etc... to allow it to happen.  Unfortunately, that means it won't be available until MET v11.0.0 beta 1 which will likely be out in mid-to-late May.

My deepest apologies for the mis-information and lack of availability of this tool.

Tara

**Comment by TaraJensen** [comment, marked as answer]
@sarahmonet - An update on this capability...  We were able to get MvMODE integrated into MET v10.1.0 and it will be released later today.  Once available, please test with your configuration file and report back as to if it works for you.  I will wait on locking this discussion until we hear back from you.  Thank you for your patience.

**Reply by sarahmonet** [reply]
Thank you! I've downloaded MET v10.1.0, however I'm still having issues with the configuration file. I have a netcdf file that I can read using the MODEconfig file but the MODEMultivarConfig gives me File "/ships19/models/sarahm/RRFS/goes16_data/2021_12_10_344/remap_GOES-16.20211210.070000.nc" of type FileType_NcMet is not an ASCII file list. Does MvMODE currently not accept netcf files?

**Reply by JohnHalleyGotway** [reply]
@sarahmonet, @TaraJensen asked me to take a look at this discussion. The usage statement for MODE has changed MET version 10.1.0... and gotten much longer! It can either be run for single-variable (i.e. as it was) or multi-variable (which is the new part).

Note that for single variable MODE "fcst_file" is the path to a gridded forecast file.
For multi-variable MODE "fcst_file" is replaced with "fcst_file_list". And that's an ASCII file containing a list of input files to be used. This [XML file](https://github.com/dtcenter/MET/blob/main_v10.1/test/xml/unit_mode_multivar.xml) contains a contrived example that runs multi-var MODE. I believe it's reading the i-th field from the i-th file.

If you're reading multiple fields from the same input file, please try listing that same input file multiple times in that ASCII fcst_file_list.

Does that produce any better results?

As you work through this, please keep track of the improvements you'd like to see in future versions of this.

```
*** Model Evaluation Tools (METV10.1.0) ***

Single Variable MODE:
=====================


Usage: mode
	fcst_file
	obs_file
	config_file
	[-config_merge merge_config_file]
	[-outdir path]
	[-log file]
	[-v level]
	[-compress level]

	where
		"fcst_file" is a gridded forecast file containing the field to be verified (required).
		"obs_file" is a gridded observation file containing the verifying field (required).
		"config_file" is a MODEConfig file containing the desired configuration settings (required).
		"-config_merge merge_config_file" overrides the default fuzzy engine settings for merging within the fcst/obs fields (optional).
		"-outdir path" overrides the default output directory (./) (optional).
		"-log file" outputs log messages to the specified file (optional).
		"-v level" overrides the default level of logging (2) (optional).
		"-compress level" overrides the compression level of NetCDF variable (optional).

Multi Variable MODE:
====================


	Usage: mode
	fcst_file_list
	obs_file_list
	config_file
	[-outdir path]
	[-log file]
	[-v level]
	where
		"fcst_file_list" is an ASCII file containing a list of forecast files to be used (required).
		"obs_file_list"  is an ASCII file containing a list of observation files to be used (required).
		"config" is a MODEConfig file containing the desired configuration settings (required).
		"-outdir path" overrides the default output directory (./) (optional).
		"-log file" outputs log messages to the specified file (optional).
		"-v level" overrides the default level of logging (2) (optional).
```

**Reply by sarahmonet** [reply]
Yes, this works. thanks!

**Reply by JohnHalleyGotway** [reply]
Great, @sarahmonet thanks for confirming. At this point, I'm going to mark this Discussion as being answered and lock conversation on it to prevent future posts.

If other issues or questions arise in your use of MODE, or any other METplus component, please feel free to start a new discussion.
